Init quantization backend config for inductor

leslie-fang-intel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #97122
* #97133
* #97132
* #96475
* #96927
* __->__ #96476

**Summary**
Init the backend config file with quantization recipes for quantization 2.0 inductor path. In this PR, we only init the recipe for `convolution` and `convolution_relu`.

**Test Plan**
```
clear && python -m pytest test_quantization.py -k test_inductor_backend_config_conv
```


cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

leslie-fang-intel committed:
Init quantization backend config for inductor

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "Init quantization backend config for inductor"




[ghstack-poisoned]

leslie-fang-intel committed:
Update on "Init quantization backend config for inductor"




[ghstack-poisoned]

leslie-fang-intel committed:
Update on "Init quantization backend config for inductor"




[ghstack-poisoned]

leslie-fang-intel committed:
Update on "Init quantization backend config for inductor"




[ghstack-poisoned]

leslie-fang-intel committed:
Update on "Init quantization backend config for inductor"




[ghstack-poisoned]

leslie-fang-intel committed:
Update on "Init quantization backend config for inductor"




[ghstack-poisoned]

leslie-fang-intel committed:
Update on "Init quantization backend config for inductor"




[ghstack-poisoned]

leslie-fang-intel committed:
Update on "Init quantization backend config for inductor"




[ghstack-poisoned]

leslie-fang-intel committed:
Update on "Init quantization backend config for inductor"




[ghstack-poisoned]

leslie-fang-intel committed:
Update on "Init quantization backend config for inductor"




[ghstack-poisoned]

leslie-fang-intel committed:
Update on "Init quantization backend config for inductor"




[ghstack-poisoned]

leslie-fang-intel committed:
Update on "Init quantization backend config for inductor"




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "Init quantization backend config for inductor"




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "Init quantization backend config for inductor"


**Summary**
Init the backend config file with quantization recipes for quantization 2.0 inductor path.

**Test Plan**
```
clear && python -m pytest test_quantization.py -k test_inductor_backend_config_conv
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "Init quantization backend config for inductor"


**Summary**
Init the backend config file with quantization recipes for quantization 2.0 inductor path.

**Test Plan**
```
clear && python -m pytest test_quantization.py -k test_inductor_backend_config_conv
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "Init quantization backend config for inductor"


**Summary**
Init the backend config file with quantization recipes for quantization 2.0 inductor path.

**Test Plan**
```
clear && python -m pytest test_quantization.py -k test_inductor_backend_config_conv
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel commented:
@jerryzh168 @supriyar @vkuzo Could you help to take a look of this PR?

leslie-fang-intel committed:
Update on "Init quantization backend config for inductor"


**Summary**
Init the backend config file with quantization recipes for quantization 2.0 inductor path. In this PR, we only init the recipe for `convolution` and `convolution_relu`.

**Test Plan**
```
clear && python -m pytest test_quantization.py -k test_inductor_backend_config_conv
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

jerryzh168 commented on test/quantization/fx/test_quantize_pt2e.py:
```diff
@@ -190,6 +192,74 @@ def forward(self, x):
 
+    @skipIfNoX86
+    @xfailIfPython311
+    def test_inductor_backend_config_conv(self):
```
can you create a new test case for this test? e.g. TestQuantizePT2EOneDNN

leslie-fang-intel replied:
Done.

jerryzh168 commented on test/quantization/fx/test_quantize_pt2e.py:
```diff
@@ -190,6 +192,74 @@ def forward(self, x):
+
+                    qconfig = get_default_qconfig("x86")
+                    qconfig_mapping = QConfigMapping().set_global(qconfig)
+                    backend_config = get_inductor_pt2e_backend_config()
```
nit: maybe get_x86_inductor_pt2e_backend_config

jerryzh168 commented on torch/ao/quantization/backend_config/_inductor_pt2e.py:
```diff
@@ -0,0 +1,45 @@
+    )
+    return conv_configs
+
+def get_inductor_pt2e_backend_config():
```
nit: maybe get_x86_inductor_pt2e_backend_config

leslie-fang-intel replied:
Thanks and Done.

jerryzh168 reviewed (approved):
LGTM, had some nit comments inline

leslie-fang-intel commented on torch/ao/quantization/backend_config/_inductor_pt2e.py:
Thanks for the suggestions. Done.

leslie-fang-intel committed:
Update on "Init quantization backend config for inductor"


**Summary**
Init the backend config file with quantization recipes for quantization 2.0 inductor path. In this PR, we only init the recipe for `convolution` and `convolution_relu`.

**Test Plan**
```
clear && python -m pytest test_quantization.py -k test_inductor_backend_config_conv
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel commented:
> LGTM, had some nit comments inline

Thanks for the comments and changed accordingly.

leslie-fang-intel committed:
Update on "Init quantization backend config for inductor"


**Summary**
Init the backend config file with quantization recipes for quantization 2.0 inductor path. In this PR, we only init the recipe for `convolution` and `convolution_relu`.

**Test Plan**
```
clear && python -m pytest test_quantization.py -k test_inductor_backend_config_conv
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "Init quantization backend config for inductor"


**Summary**
Init the backend config file with quantization recipes for quantization 2.0 inductor path. In this PR, we only init the recipe for `convolution` and `convolution_relu`.

**Test Plan**
```
clear && python -m pytest test_quantization.py -k test_inductor_backend_config_conv
```


cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10

[ghstack-poisoned]

leslie-fang-intel commented:
@pytorchbot merge
